### PR TITLE
feat: Picker - Item description support

### PR DESCRIPTION
**Pull request by bmingles**
- Normalized keys for item / section now remain undefined if not provided. Same for item `textValue`
- Proper rendering of items with descriptions
- Fixed tooltips for items with descriptions. Also changed default placement to `right`
- isElementOfType util
- Added SpectrumThemeProvider to Popper

Supports deephaven/deephaven-plugins/issues/293

### CODE REVIEW

File: packages/components/src/spectrum/picker/PickerItemContent.tsx [hunk 1]
```diff
@@ -12,25 +14,57 @@ export interface PickerItemContentProps {
...
+            // content. The index suffix is an extra precation for when 2 <Text>
+            // nodes have the same value.
+            key: item.key ?? `${item.props.children}_${i}`,
+            UNSAFE_className: cl(
+              item.props.UNSAFE_className,
+              stylesCommon.spectrumEllipsis
+            ),
+          })
+        : item
+    );
```
Comment (mofojed) [inline packages/components/src/spectrum/picker/PickerItemContent.tsx:59]: Ah, when we were talking about this, I didn't realize the children we were talking about were _inside_ the picker items - i.e. there's no need for a key.

You should be able to get around this warning by just using `React.Children` instead:
```
    content = React.Children.map(content, (element, i) =>
      isElementOfType(element, Text)
        ? cloneElement(element, {
            ...element.props,
            UNSAFE_className: cl(
              element.props.UNSAFE_className,
              stylesCommon.spectrumEllipsis
            ),
          })
        : element
    );
```

File: packages/components/src/spectrum/picker/PickerUtils.ts [hunk 2]
```diff
@@ -120,11 +120,11 @@ export function isNormalizedPickerSection(
  * @param itemOrSection The picker item or section
  * @returns A `PickerItemKey` for the picker item
  */
-function normalizeItemKey(item: PickerItem): PickerItemKey;
-function normalizeItemKey(section: PickerSection): Key;
+function normalizeItemKey(item: PickerItem): PickerItemKey | undefined;
```
Comment (mofojed) [inline packages/components/src/spectrum/picker/PickerUtils.ts:123]: I'd update the doc `@returns` as to when it returns `undefined`, as this is unexpected from the current docs/name.

**Review by mofojed — CHANGES_REQUESTED** [review]

**Comment by mofojed** [comment]
Need to update e2e snapshots as well.

File: packages/components/src/spectrum/picker/PickerItemContent.tsx (hunk 1 above)
Comment (bmingles) [reply packages/components/src/spectrum/picker/PickerItemContent.tsx:59]: Ah nice, glad you knew about this one. It felt gross the way I implemented it.

File: packages/code-studio/src/styleguide/SamplesMenu.tsx [hunk 3]
```diff
@@ -56,7 +56,7 @@ export function SamplesMenu(): JSX.Element {
     );
 
     const spectrumComparisonSamples = document.querySelector(
-      `#${SPECTRUM_COMPARISON_SAMPLES_ID}`
+      `#sample-section-${SPECTRUM_COMPARISON_SAMPLES_ID}`
```
Comment (bmingles) [inline packages/code-studio/src/styleguide/SamplesMenu.tsx:59]: This was a bug causing duplicate ids and the menu would always navigate to the first occurrence

**Review by mofojed — APPROVED** [review]

**Pull request merged**